src/graphics/program-lib/programs/standard.js

```diff
@@ -1339,11 +1339,20 @@ var standard = {
         code = this._fsAddStartCode(code, device, chunks, options);
 
         if (needsNormal) {
-            if (options.twoSidedLighting) {
-                code += "   dVertexNormalW = gl_FrontFacing ? vNormalW * twoSidedLightingNegScaleFactor : -vNormalW * twoSidedLightingNegScaleFactor;\n";
+            if (!options.hasTangents && device.extStandardDerivatives) {
+                if (options.twoSidedLighting) {
+                    code += "   dVertexNormalW = normalize(gl_FrontFacing ? vNormalW * twoSidedLightingNegScaleFactor : -vNormalW * twoSidedLightingNegScaleFactor);\n";
+                } else {
+                    code += "   dVertexNormalW = normalize(vNormalW);\n";
+                }
             } else {
-                code += "   dVertexNormalW = vNormalW;\n";
+                if (options.twoSidedLighting) {
+                    code += "   dVertexNormalW = gl_FrontFacing ? vNormalW * twoSidedLightingNegScaleFactor : -vNormalW * twoSidedLightingNegScaleFactor;\n";
+                } else {
+                    code += "   dVertexNormalW = vNormalW;\n";
+                }
```

Comment:
why don't we normalize here as well .. are these normalized already?

Reply:
they get normalized or not based on fastTbn option - actually... we may want to use the fastTbn option to skip normalization even in the case where there are no tangents - I shall make that change.

Reply:
I added a change to respect the fastTbn material option - if the fastTbn option is true then the normalize is not performed